creator-node/src/services/sync/secondarySyncFromPrimary.js

```diff
@@ -49,6 +50,24 @@ const handleSyncFromPrimary = async ({
     }
   }
 
+  // Ensure this node is syncing from the user's primary
+  const userReplicaSet = [
+    ...new Set(
```

Comment:
o shit i just realized - does doing this array -> set -> array conversion preserve order? 😬

Reply:
also if we remove duplicates here how do we know if it removes first one or second one
i.e. what if [A, B, A] becomes [B, A]

Reply:
as per our sync up, I made a new ReplicaSet type and util to convert a `creator_node_endpoint` array to it + changed all usages to consume this

Reply:
wow dude